src/_tkagg.cpp

```diff
@@ -190,11 +190,17 @@ void load_tkinter_funcs(void)
         PyErr_SetString(PyExc_RuntimeError, dlerror());
         goto exit;
     }
-    load_tk(tkinter_lib);
-    // dlclose is safe because tkinter has been imported.
-    dlclose(tkinter_lib);
-    goto exit;
+    if (load_tk(tkinter_lib)) {
+        goto exit;
+    }
+
 exit:
+    // We don't need to keep a reference open as the main program & tkinter
+    // have been imported.
+    if ((main_program && dlclose(main_program))
+        || (tkinter_lib && dlclose(tkinter_lib))) {
```

Comment:
This will short-circuit and not close `tkinter_lib` if `main_program` doesn't close.

Reply:
Indeed, using a bitwise (non-shortcircuiting) "or" should work better.
(In practice if we fail to close main_program we're likely in deep sh\*t but heh.)